packages/amplify-graphql-searchable-transformer/src/graphql-searchable-transformer.ts

```diff
@@ -0,0 +1,257 @@
+  makeSearchableXSortInputObject,
+} from './definitions';
+import assert from 'assert';
+import { setMappings } from './cdk/create-layer-cfnMapping';
+import { createEsDomain, createEsDomainRole } from './cdk/create-es-domain';
+import { createEsDataSource } from './cdk/create-es-datasource';
+import { createEventSourceMapping, createLambda, createLambdaRole } from './cdk/create-streaming-lambda';
+import { createStackOutputs } from './cdk/create-cfnOutput';
+
+const nonKeywordTypes = ['Int', 'Float', 'Boolean', 'AWSTimestamp', 'AWSDate', 'AWSDateTime'];
+const STACK_NAME = 'SearchableStack';
+export class SearchableModelTransformer extends TransformerPluginBase {
+  searchableObjectTypeDefinitions: { node: ObjectTypeDefinitionNode; fieldName: string }[];
+  constructor() {
+    super(
+      'amplify-searchable-transformer',
+      /* GraphQL */ `
+        directive @searchable(queries: SearchableQueryMap) on OBJECT
+        input SearchableQueryMap {
+          search: String
+        }
+      `,
+    );
+    this.searchableObjectTypeDefinitions = [];
+  }
+
+  generateResolvers = (context: TransformerContextProvider): void => {
+    const { Env } = ResourceConstants.PARAMETERS;
+
+    const { HasEnvironmentParameter } = ResourceConstants.CONDITIONS;
+
+    const stack = context.stackManager.createStack(STACK_NAME);
+
+    setMappings(stack);
+
+    const envParam = context.stackManager.getParameter(Env) as CfnParameter;
+
+    new CfnCondition(stack, HasEnvironmentParameter, {
+      expression: Fn.conditionNot(Fn.conditionEquals(envParam, 'NONE')),
+    });
+
+    const isProjectUsingDataStore = false;
+
+    stack.templateOptions.description = 'An auto-generated nested stack for searchable.';
+    stack.templateOptions.templateFormatVersion = '2010-09-09';
+
+    const parameterMap = createParametersInStack(stack);
+
+    const domain = createEsDomain(stack, parameterMap, context.api.apiId);
+
+    const elasticsearchRole = createEsDomainRole(stack, parameterMap, context.api.apiId, envParam);
+
+    domain.grantReadWrite(elasticsearchRole);
+
+    const datasource = createEsDataSource(
+      stack,
+      context.api,
+      domain.domainEndpoint,
+      elasticsearchRole,
+      stack.parseArn(domain.domainArn).region,
+    );
+
+    // streaming lambda role
+    const lambdaRole = createLambdaRole(stack, parameterMap);
+    domain.grantWrite(lambdaRole);
+
+    // creates streaming lambda
+    const lambda = createLambda(
+      stack,
+      context.api,
+      parameterMap,
+      lambdaRole,
+      domain.domainEndpoint,
+      isProjectUsingDataStore,
+      stack.parseArn(domain.domainArn).region,
+    );
+
+    for (const def of this.searchableObjectTypeDefinitions) {
+      const type = def.node.name.value;
+      const typeName = context.output.getQueryTypeName();
+      const table = getTable(context, def.node);
+      const ddbTable = table as Table;
+      ddbTable.grantStreamRead(lambdaRole);
+      assert(ddbTable);
```

Comment:
It looks like we're asserting `ddbTable` after already using it. Is that intentional?